### ISSUE TITLE
Support confirm password-type validation 

### ISSUE THREAD

**Issue by kentcdodds** [brophdawg11/remix-validity-state#10]
I have a "confirm password" field that needs to be validated against the password field. Can custom validators get all the form fields?

**Comment by brophdawg11** [comment]
Hm, yeah I think password is an interesting case here.  In the past with an approach like this (client-side only) I used the presence of the attribute as the indicator to validate (just like built-in validations) and then you could pass additional info as the attribute value which would let you validate based on an external value.

So you would do something like the following:

```js
let validations = {
  dataIncludedIn: (inputValue, attrValue) {
    return attrValue.split(',').includes(inputValue)
  }
}
```

```html
// Note the attribute value here can come from any other spot - another input value, 
// controlled state, etc.
<input name="framework" data-included-in="remix,vue,preact,solid,svelte" />
```

The symmetry with a validation attribute like `minlength="5"` was always super nice here IMO.  

However, this approach doesn't work in remix with the network chasm since we don't have access to the attribute values on the server, only the input values.  Nor would it be a good idea for password confirmation since you'd be putting the plaintext password in a data attr 😬 

So yeah, I think changing the signature of custom validation functions to `validate(inputValue, formValues)` makes sense to solve this password case 👍

**Comment by kentcdodds** [comment]
Maybe it shouldn't be formValues, but FormData? To account for multiple fields with the same name as in #8?

**Comment by brophdawg11** [comment]
yeah agreed - this is what it looks like now in my local - should have something released soon! 

Note `FormData` can be `undefined` in the custom validator since the `input` can be used standalone not inside a form, although that sort of defeats the purpose :)

```js
const formValidations = {
  password: {
    type: "password",
    required: true,
  },
  confirmPassword: {
    type: "password",
    required: true,
    matches(value, formData) {
      return value === formData?.get("password");
    },
  },
};

const customErrorMessages = {
 matches: () => `Passwords don't match!`,
};
```

**Comment by kentcdodds** [comment]
:chefs-kiss:

**Comment by brophdawg11** [comment]
Released inc `0.4.0`
